RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitConvertGPU2CPUFormat.cc

```diff
@@ -0,0 +1,123 @@
+// framework
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h" 
+
+// algorithm specific
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CUDADataFormats/EcalRecHitSoA/interface/EcalUncalibratedRecHit_soa.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "RecoLocalCalo/EcalRecAlgos/interface/Common.h"
+
+#include <iostream>
+
+class EcalUncalibRecHitConvertGPU2CPUFormat
+    : public edm::stream::EDProducer<>
+{
+public:
+    explicit EcalUncalibRecHitConvertGPU2CPUFormat(edm::ParameterSet const& ps);
+    ~EcalUncalibRecHitConvertGPU2CPUFormat() override;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+    using GPURecHitType = ecal::UncalibratedRecHit<ecal::Tag::soa>;
+    void produce(edm::Event&, edm::EventSetup const&) override;
+
+private:
+    edm::EDGetTokenT<ecal::SoAUncalibratedRecHitCollection> recHitsGPUEB_;
+    edm::EDGetTokenT<ecal::SoAUncalibratedRecHitCollection> recHitsGPUEE_;
+
+    std::string recHitsLabelCPUEB_, recHitsLabelCPUEE_;
+};
+
+void EcalUncalibRecHitConvertGPU2CPUFormat::fillDescriptions(
+        edm::ConfigurationDescriptions& confDesc) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<edm::InputTag>("recHitsLabelGPUEB", 
+        edm::InputTag("ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEB"));
+    desc.add<edm::InputTag>("recHitsLabelGPUEE", 
+        edm::InputTag("ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEE"));
+
+    desc.add<std::string>("recHitsLabelCPUEB", "EcalUncalibRecHitsEB");
+    desc.add<std::string>("recHitsLabelCPUEE", "EcalUncalibRecHitsEE");
+
+    std::string label = "ecalUncalibRecHitConvertGPU2CPUFormat";
+    confDesc.add(label, desc);
+}
+
+EcalUncalibRecHitConvertGPU2CPUFormat::EcalUncalibRecHitConvertGPU2CPUFormat(
+        const edm::ParameterSet& ps) 
+{
+    recHitsGPUEB_ = consumes<ecal::SoAUncalibratedRecHitCollection>(
```

Comment:
can you move these to the initialiser list ?
E.g.
```c++
EcalUncalibRecHitConvertGPU2CPUFormat::EcalUncalibRecHitConvertGPU2CPUFormat(const edm::ParameterSet& ps) :
    recHitsGPUEB_(consumes<ecal::SoAUncalibratedRecHitCollection>(...)),
    ...
```

Reply:
done